plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/standard/StandardDeployCommandHandler.java

```diff
@@ -41,22 +49,25 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
     try {
       IProject project = helper.getProject(event);
       if (project != null) {
-        launchDeployJob(project);
+        launchDeployJob(project, new SameShellProvider(HandlerUtil.getActiveShell(event)));
```

Comment:
I think the deploy job should be written as a headless job and all parameters should be obtained here, in this handler, and configured on the job.  So that includes obtaining the login credential, the console stream.  The job itself should not change any state (e.g., the login state) as that's a UI thing.